src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -181,22 +138,12 @@ void remove_virtual_functionst::remove_virtual_function(
   goto_programt new_code_calls;
   goto_programt new_code_gotos;
 
-  // Get a pointer from which we can extract a clsid.
-  // If it's already a pointer to an object of some sort, just use it;
-  // if it's void* then use the parent of all possible candidates,
-  // which by the nature of get_functions happens to be the last candidate.
-
   exprt this_expr=code.arguments()[0];
-  assert(this_expr.type().id()==ID_pointer &&
-         "Non-pointer this-arg in remove-virtuals?");
-  const auto &points_to=this_expr.type().subtype();
-  if(points_to==empty_typet())
-  {
-    symbol_typet symbol_type(functions.back().class_id);
-    this_expr=typecast_exprt(this_expr, pointer_typet(symbol_type));
-  }
-  exprt deref=dereference_exprt(this_expr, this_expr.type().subtype());
-  exprt c_id2=build_class_identifier(deref);
+  // If necessary, cast to the last candidate function to get the object's clsid.
```

Comment:
Just had a look at the linter's complaints: it seems this line is too long; could you please also add a commit to address the other 5 concerns on this file, even though they aren't your changes:
```
src/goto-programs/remove_virtual_functions.cpp:142:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
src/goto-programs/remove_virtual_functions.cpp:199:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
src/goto-programs/remove_virtual_functions.cpp:199:  Statement after an if should be on a new line  [readability/braces] [5]
src/goto-programs/remove_virtual_functions.cpp:200:  Statement after an if should be on a new line  [readability/braces] [5]
src/goto-programs/remove_virtual_functions.cpp:305:  Statement after an if should be on a new line  [readability/braces] [5]
src/goto-programs/remove_virtual_functions.cpp:383:  Should have a space between // and comment  [whitespace/comments] [4]
```